modules/navier_stokes/include/linearfvkernels/LinearFVScalarAdvection.h

```diff
@@ -0,0 +1,55 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "LinearFVFluxKernel.h"
+#include "RhieChowMassFlux.h"
+#include "LinearFVAdvectionDiffusionBC.h"
+
+/**
+ * An advection kernel that implements the advection term for the enthalpy in the
+ * energy equation.
+ */
+class LinearFVScalarAdvection : public LinearFVFluxKernel
+{
+public:
+  static InputParameters validParams();
+  LinearFVScalarAdvection(const InputParameters & params);
+
+  virtual Real computeElemMatrixContribution() override;
+
+  virtual Real computeNeighborMatrixContribution() override;
+
+  virtual Real computeElemRightHandSideContribution() override;
+
+  virtual Real computeNeighborRightHandSideContribution() override;
+
+  virtual Real computeBoundaryMatrixContribution(const LinearFVBoundaryCondition & bc) override;
+
+  virtual Real computeBoundaryRHSContribution(const LinearFVBoundaryCondition & bc) override;
+
+  virtual void setupFaceData(const FaceInfo * face_info) override;
+
+protected:
+  /// The Rhie-Chow user object that provides us with the face velocity
+  const RhieChowMassFlux & _mass_flux_provider;
```

Comment:
Maybe we can just divide the mass flux by the face rho. Either in the RC object or in the kernel.